source/gmsh/utilities.cc

```diff
@@ -54,12 +54,18 @@ namespace Gmsh
                                            const AdditionalParameters &prm)
   {
     std::string base_name = prm.output_base_name;
+    int file_descriptor = -1;
     if (base_name == "")
-      base_name = std::tmpnam(nullptr);
+      {
+        // generate temporary file name
+        base_name = "tmp-XXXXXX";
+        file_descriptor = mkstemp(&base_name[0]);
```

Comment:
Can we use `std::tmpfile` instead? I fear that using posix `mkstemp` unguarded might not be portable.